frontend/src/layout/navigation-3000/navigationLogic.ts

```diff
@@ -117,8 +128,62 @@ export const navigation3000Logic = kea<navigation3000LogicType>([
                 }),
             },
         ],
+        newItemInlineCategory: [
+            null as string | null,
+            {
+                initiateNewItemInlineInCategory: (_, { category }) => category,
+                saveNewItemComplete: () => null,
+                cancelNewItem: () => null,
+                toggleSidebar: () => null,
+                showSidebar: () => null,
+                hideSidebar: () => null,
+            },
+        ],
+        savingNewItem: [
+            false,
+            {
+                saveNewItem: () => true,
+                saveNewItemComplete: () => false,
+            },
+        ],
     }),
     listeners(({ actions, values }) => ({
+        initiateNewItemInCategory: ({ category: categoryKey }) => {
+            const category = values.activeNavbarItem?.logic.values.contents?.find((item) => item.key === categoryKey)
+            if (!category) {
+                throw new Error(`Sidebar category '${categoryKey}' doesn't exist`)
+            } else if (!category.onAdd || typeof category.onAdd !== 'function') {
+                throw new Error(`Sidebar category '${categoryKey}' doesn't support onAdd`)
+            }
+            if (category.onAdd.length === 0) {
```

Comment:
I never knew you could use `.length` to check for number of arguments in JS - learned something new today.

Reply:
You can 😄 #JustJSThings